src/cpp/src/visual_language/inputs_embedder.hpp

```diff
@@ -67,6 +67,7 @@ class InputsEmbedder {
 private:
     class IInputsEmbedder;
     std::shared_ptr<IInputsEmbedder> m_impl;
+    std::mutex m_inputs_embedder_mutex;
```

Comment:
maybe we need to move a mutex to add_request with rgb images then ?

Reply:
Agree, moved to add_request with rgbs.